pkg/controller/kieapp/kieapp_controller.go

```diff
@@ -80,6 +80,9 @@ func (reconciler *Reconciler) Reconcile(request reconcile.Request) (reconcile.Re
 		reconciler.setFailedStatus(instance, api.UnknownReason, err)
 		return reconcile.Result{}, err
 	}
+	if instance.GetDeletionTimestamp() != nil {
```

Comment:
If the resource is being deleted just don't reconcile. I don't think it is necessary, otherwise, resources deleted by the Finalizers might be re-created.

Reply:
did you test this in a cluster?

Reply:
I don't think we need this here